package_update/package_update.py

```
@@ -37,6 +41,109 @@
+
+@with_semaphore(async_sem)
+async def _check_package_availability(
+    package: Tuple[str, str, str],
+    sources: Dict[str, Any],
+    removed_packages: set,
+    missing_package: MissingPackageMessage,
+) -> bool:
+    src = sources[package[1]]
+    if not package[0] in src["packages"]:
+        removed_packages.add((package[1], package[0]))
+        print((package[1], package[0]))
+        try:
+            await missing_package.publish_to_topic(missing_package.MessageContents(
+                index_url=package[1],
+                package_name=package[0],
+            ))
+            _LOGGER.info("%r no longer provides %r", package[1], package[0])
+            return False
+        except Exception as e:
+            _LOGGER.exception("Failed to publish with the following error message: %r", e)
+    return True
+
+@with_semaphore(async_sem)
+async def _check_hashes(
+    package_version: Tuple[str, str, str],
+    package_versions,
+    source,
+    removed_packages: set,
+    missing_version: MissingVersionMessage,
+    hash_mismatch: HashMismatchMessage,
+    graph: GraphDatabase,
+) -> bool:
+    if not package_version[1] in package_versions.versions:
+        try:
+            await missing_version.publish_to_topic(
+                missing_version.MessageContents(
+                    index_url=package_version[2], package_name=package_version[0], package_version=package_version[1],
+                ),
+            )
+            _LOGGER.info("%r no longer provides %r-%r", package_version[2], package_version[0], package_version[1])
+            return False
+        except Exception as identifier:
+            _LOGGER.exception("Failed to publish with the following error message: %r", str(identifier))
+
+    try:
+        source_hashes = {i["sha256"] for i in await source.get_package_hashes(package_version[0], package_version[1])}
+    except ClientResponseError:
+        return False  # webpage might be down
```

Can the exception be logged ussing `_LOGGER.exception` instead of silently passed? Here and other places as well.